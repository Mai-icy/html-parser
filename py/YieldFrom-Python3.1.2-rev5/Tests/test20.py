#
#   Test returning value from delegated 'throw'
#

def g1():
	try:
		print("Starting g1")
		yield "g1 ham"
		yield from g2()
		yield "g1 eggs"
	finally:
		print("Finishing g1")

def g2():
	try:
		print("Starting g2")
		yield "g2 spam"
		yield "g2 more spam"
	except LunchError:
		print("Caught LunchError in g2")
		yield "g2 lunch saved"
		yield "g2 yet more spam"

class LunchError(Exception):
	pass

g = g1()
for i in range(2):
	x = next(g)
	print("Yielded", x)
e = LunchError("tomato ejected")
g.throw(e)
for x in g:
	print("Yielded", x)
