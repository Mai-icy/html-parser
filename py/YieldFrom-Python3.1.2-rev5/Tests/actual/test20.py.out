Starting g1
Yielded g1 ham
Starting g2
Yielded g2 spam
Caught LunchError in g2
Yielded g2 yet more spam
Yielded g1 eggs
Finishing g1
